File: Chapter 03/C#/3.2 Coupling Exercise/After/UserReportGenerator.cs
namespace Chapter03.CouplingTest.After;

public class UserReportGenerator
{
    private readonly UserDataService _dataService = new();

    public string GenerateReport(int userId)
    {
        // Low Coupling: The client just asks for what it needs once
        var report = _dataService.GetUserReport(userId);
        
        return $"User Report for {report.Name} ({report.Email}) - Total Spent: ${report.TotalSpent}";
    }
}

File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/3.2 Coupling Exercise/After/UserReportGenerator.java
package com.grokkingsoftwarearchitecture.chapter03.couplingexercise.after;

public class UserReportGenerator {
    private final UserDataService dataService = new UserDataService();

    public String generateReport(int userId) {
        UserReportData report = dataService.getUserReport(userId);
        return String.format("User Report for %s (%s) - Total Spent: $%.2f", report.name, report.email, report.totalSpent);
    }
}